hadoop-project/pom.xml

```diff
@@ -2428,9 +2428,10 @@
       </activation>
       <properties>
         <hbase.version>${hbase.two.version}</hbase.version>
-        <hbase-compatible-hadoop.version>3.0.0</hbase-compatible-hadoop.version>
+        <hbase-compatible-hadoop.version>2.8.5</hbase-compatible-hadoop.version>
```

Comment:
how is this upgrade downgrading the hadoop version? Earlier on a lower version we were good with 3.x and now to upgrade we are coming down. 
Out of curiosity, why 2.8.5, nothing from 2.10 line or so?

Reply:
Valid question. The earlier hbase version 2.0 is not only not so stable but also it just worked with Hadoop 3 version magically but as we upgrade HBase 2.x versions, they won't work by default with Hadoop 3 because as of today, hbase has only published hadoop 2 compatible hbase 2 artifacts by default. If we want to make hbase 2 work with hadoop 3, first we need to compile hbase 2 with hadoop-3.0 profile, only then we get hadoop 3 compatible hbase 2 jars. Hence, we should have anyways kept `hbase-compatible-hadoop.version` on Hadoop's 2.x release line but somehow we didn't and this worked with HBase 2.0.

I have provided detailed explanation of this also on HADOOP-17668. Comment [here](https://issues.apache.org/jira/browse/HADOOP-17668?focusedCommentId=17454172&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-17454172) (hope I covered the right link, Jira interface seems to have been updated recently)

And yes @jojochuang's review would also be great, thanks for the reminder (my bad, I missed).

Reply:
hmm, I need to go through this and try, considering @aajisaka is convinced. I will back off for now. 
You guys can proceed. Nothing blocking from my side

Reply:
> Out of curiosity, why 2.8.5, nothing from 2.10 line or so?

To provide in-depth dependency view for this question: https://github.com/apache/hbase/blob/rel/2.2.4/pom.xml#L1373-L1378
```
    <hadoop-two.version>2.8.5</hadoop-two.version>
    <hadoop-three.version>3.1.2</hadoop-three.version>
    <!-- These must be defined here for downstream build tools that don't look at profiles.
         They ought to match the values found in our default hadoop profile, which is
         currently "hadoop-2.0". See HBASE-15925 for more info. -->
    <hadoop.version>${hadoop-two.version}</hadoop.version>
```

Reply:
Ah i see. Yes by default HBase 2 is built with Hadoop 2 profile. It is not until HBase 3.0 (being developed) that Hadoop 2 profile got removed.